### PR TITLE
[5.8] Provides rule support for ANSI character length validation

### PR DESCRIPTION
**Pull request by kaysonwu**
Provide validation for multi-byte string length.

Application scenario：
Determine the length of the Chinese character, 1 Chinese character is 2 lengths.

### CODE REVIEW

**Comment by sisve** [comment]
This doesn't look related to ANSI (or Windows 1252) at all. Could you clarify how ansi is involved in this?

Your description talks about lengths of Chinese characters, but the mb_strwidth is about the _width_ of a character. I believe this to be a translation detail, but we'll need to use the correct terms when writing the documentation.

**Comment by kaysonwu** [comment]
Of course, I'm happy to share my opinion. First, laravel already has various rules for length validation. So, using `strwidth` will conflict with they.  Then is it more appropriate to name it based on the string type?

`ansi`  is a character code. to enable a computer to support more languages, usually 1 byte in the range `0x00~0x7f` is used to represent 1 character, 2 bytes in the range `0x80~0xFFFF` to represent 1 character.

`DBCS`(double-byte character set)  also appears in my alternative，but it seems that it can't fully represent `strwidth`.

In summary, I personally think that `ansi` is suitable, but if there is a better alternative, I would be happy to accept.

**Comment by sisve** [comment]
ANSI is the American National Standards Institute, but there's a character encoding commonly called "ANSI" that is really Windows-1252, which in turn is very close to ISO-8859-1. These are single-byte encodings, they only supports up to 256 characters (usually slightly less due to undefined character values).

When you write "1 byte in the range 0x00~0x7f is used to represent 1 character, 2 bytes in the range 0x80~0xFFFF to represent 1 character", then it sounds more like a variable-length encoding like UTF-8. (Slightly irrelevant note; while UTF-8 and mb_strlen works with codepoints, and we [erroneously] assume that 1 codepoint == 1 character in the framework.)

Now, the mb_strwidth function is about half-width and full-width characters. Could we rename the flag "use_width" instead of ansi?

**Comment by kaysonwu** [comment]
This is a good suggestion, I am going to modify it.

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Comment by kaysonwu** [comment]
@taylorotwell 

First，can't express this idea more elegantly by extending the rules. Because laravel's length validation rules are too perfect.

Rule name may not be laravel style. But the function is very practical.  So, if you could give me a name, I would appreciate it very much. 

Why is it practical?

1. fix  ` $hasNumeric = $this->hasRule($attribute, $this->numericRules);` statement must be executed when getting the attribute size.
2. meet the needs of the actual length of double-byte characters.
